cloudinary-core/src/main/java/com/cloudinary/strategies/AbstractApiStrategy.java

```diff
@@ -31,4 +32,19 @@ protected String createApiUrl (Iterable<String> uri, String prefix, String cloud
     public abstract ApiResponse callApi(HttpMethod method, Iterable<String> uri, Map<String, ? extends Object> params, Map options) throws Exception;
 
     public abstract ApiResponse callAccountApi(HttpMethod method, Iterable<String> uri, Map<String, ? extends Object> params, Map options) throws Exception;
+
+    protected String getAuthorizationHeaderValue(String apiKey, String apiSecret, String oauthToken) {
+        if (oauthToken != null){
+            return "Bearer " + oauthToken;
+        } else {
+            return "Basic " + Base64Coder.encodeString(apiKey + ":" + apiSecret);
+        }
+    }
+
+    protected void validateAuthorization(String apiKey, String apiSecret, String oauthToken) {
+        if (oauthToken == null) {
+            if (apiKey == null) throw new IllegalArgumentException("Must supply api_key");
+            if (apiSecret == null) throw new IllegalArgumentException("Must supply api_secret");
+        }
```

Comment:
what if all of them are null?

Reply:
If all are null then this is thrown - `IllegalArgumentException("Must supply api_key"). Same behaviour as before for people not using oauth so it doesn't break anything.